controllers/managedocs_controller.go

```diff
@@ -722,6 +731,39 @@ func (r *ManagedOCSReconciler) getDesiredConvergedStorageCluster() (*ocsv1.Stora
 	return sc, nil
 }
 
+func (r *ManagedOCSReconciler) getDesiredConsumerStorageCluster() (*ocsv1.StorageCluster, error) {
+
+	capacityAsString := r.addonParams[consumerRequestedCapacity]
+	onboardingTicket := r.addonParams[consumerOnboardingTicket]
+	storageProviderEndpoint := r.addonParams[consumerStorageProviderEndpoint]
+
+	requestedCapacity, err := resource.ParseQuantity(capacityAsString)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid storage cluster capacity value: %v", capacityAsString)
+	}
+
+	currentState := r.storageCluster.Spec.ExternalStorage
+	sc := templates.ConsumerStorageClusterTemplate.DeepCopy()
+	es := &sc.Spec.ExternalStorage
+
+	if currentState.OnboardingTicket != onboardingTicket {
+		es.OnboardingTicket = onboardingTicket
+	}
+
+	if currentState.StorageProviderEndpoint != storageProviderEndpoint {
+		es.StorageProviderEndpoint = storageProviderEndpoint
+	}
+
+	if currentState.RequestedCapacity == nil || currentState.RequestedCapacity.Cmp(requestedCapacity) != -1 {
+		es.RequestedCapacity = &requestedCapacity
+	} else {
+		r.Log.V(-1).Info("Requested storage capacity will result in downscaling, which is not supported. Skipping")
+		es.RequestedCapacity = currentState.RequestedCapacity
+	}
```

Comment:
Can we have a separate function to prevent downscaling because it will be done for consumer, provider, and converged?

Reply:
- I feel in-lining the way it is good for now as at-least three arguments should be sent for one validation
- Just a thought, if another reviewer weighs in for moving it to a function, then it'd be fine.

Reply:
Not relevant - see my last comment